astropy/units/structured.py

```diff
@@ -5,17 +5,37 @@
 
 # Standard library
 import operator
+from typing import Any
 
 import numpy as np
 
 from .core import UNITY, Unit, UnitBase
 
-__all__ = ['StructuredUnit']
+__all__ = ['StructuredUnit', "is_structured"]
 
 
 DTYPE_OBJECT = np.dtype('O')
 
 
+def is_structured(x: Any, /) -> bool:
```

Comment:
where would be a better place for this function, somewhere in ``utils`` I think, but no where is immediately obvious.